src/ucp/core/ucp_am.c

```diff
@@ -771,14 +840,21 @@ ucp_am_try_send_short(ucp_ep_h ep, uint16_t id, uint32_t flags,
                       const void *header, size_t header_length,
                       const void *buffer, size_t length)
 {
-    if (ucs_likely(!(flags & UCP_AM_SEND_REPLY) &&
-                   ((length == 0) || (header_length == 0)) &&
-                   ((ssize_t)(length + header_length) <=
-                    ucp_ep_config(ep)->am.max_short))) {
+    if (ucs_unlikely(((length != 0) && (header_length != 0)) ||
```

Comment:
looks weird, why we refuse to short-send header+data?
we can copy both to the send buffer..

Reply:
several reasons:
1) we recommend users send just header for latency sensitive data
2) simpler code, less branches

Reply:
IMO it could be simpler to check if short is supported: can just check if header+len<=max_short

Reply:
yes, but
1) I' ll have to limit short protocol by `MAX_ALLOCA` size even for non-reply protocol (which significantly affects performance if we have MEMIC)
2)  I'll have to do something like below anyway
```
if ((length != 0) && (header_length != 0))
    memcpy(buf, ...);
} else if (length !=0) {
    buf = payload;
} else {
    buf = header;
}
```

Reply:
ok, i assume we get rid of it anyway after adding am short iov?

Reply:
yes, will add TODO about it